File: Merge_2D_Sum/merge.cs
using System.Collections.Generic;

public class Solution
{
    public int[][] MergeArrays(int[][] nums1, int[][] nums2)
    {
        List<int[]> result = new List<int[]>();

        int i = 0,
            j = 0;

        while (i < nums1.Length && j < nums2.Length)
        {
            if (nums1[i][0] == nums2[j][0])
            {
                int key = nums1[i][0];
                int sum = nums1[i][1] + nums2[j][1];
                result.Add(new int[] { key, sum });
                i++;
                j++;
            }
            else if (nums1[i][0] > nums2[j][0])
            {
                result.Add(nums2[j]);
                j++;
            }
            else
            {
                result.Add(nums1[i]);
                i++;
            }
        }

        while (i < nums1.Length)
        {
            result.Add(nums1[i]);
            i++;
        }

        while (j < nums2.Length)
        {
            result.Add(nums2[j]);
            j++;
        }

        return result.ToArray();
    }
}

File: Merge_2D_Sum/merge.py
from typing import List


class Solution:
    def mergeArrays(
        self, nums1: List[List[int]], nums2: List[List[int]]
    ) -> List[List[int]]:
        merge: List[List[int]] = []

        i, j = 0, 0

        while i < len(nums1) and j < len(nums2):
            if nums1[i][0] == nums2[j][0]:
                sum = nums1[i][1] + nums2[j][1]
                merge.append([nums1[i][0], sum])
                i, j = i + 1, j + 1
            elif nums1[i][0] > nums2[j][0]:
                merge.append(nums2[j])
                j += 1
            else:
                merge.append(nums1[i])
                i += 1

        while i < len(nums1):
            merge.append(nums1[i])
            i += 1

        while j < len(nums2):
            merge.append(nums2[j])
            j += 1

        return merge
